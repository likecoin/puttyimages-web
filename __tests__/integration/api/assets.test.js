const api = require('./api');

test('get asset info', async () => {
  const response = await api.get(
    'assets/04f01eb90f7be1b1628218c5a9f1f87c5704bc79548f15673695a3e7264ba88b'
  );
  expect(response.data).toMatchSnapshot();
});

test('get featured assets', async () => {
  const response = await api.get('assets/featured/list');
  expect(response.data).toMatchSnapshot();
});

test('update asset view count by 1', async () => {
  const assetId =
    '04f01eb90f7be1b1628218c5a9f1f87c5704bc79548f15673695a3e7264ba88b';
  await api.post(`assets/${assetId}/view`);
  const response = await api.get(`assets/${assetId}`);
  const { viewCount } = response.data.stats;

  expect(viewCount).toBe(1);
});

test('update asset download count by 1', async () => {
  const assetId =
    '04f01eb90f7be1b1628218c5a9f1f87c5704bc79548f15673695a3e7264ba88b';
  await api.post(`assets/${assetId}/download`);
  const response = await api.get(`assets/${assetId}`);
  const { downloadCount } = response.data.stats;

  expect(downloadCount).toBe(1);
});
