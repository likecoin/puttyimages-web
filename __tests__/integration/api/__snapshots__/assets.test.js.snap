// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get asset info 1`] = `
Object {
  "createdAt": "2018-06-20T00:00:00.000Z",
  "description": "avocado",
  "fingerprint": "04f01eb90f7be1b1628218c5a9f1f87c5704bc79548f15673695a3e7264ba88b",
  "height": 15,
  "ipfs": "QmfEcrSyPS3K7WNRZLseGu3TqSpQqRToJsvT4svUe1UDmd",
  "ipld": "zdpuAqoWEAALng3FeUcF8GY66UsK17wd3fGxr9XbXu6baoKLN",
  "license": "cc0",
  "like": Object {
    "count": 0,
  },
  "tags": Array [
    Object {
      "name": "avocado",
    },
    Object {
      "name": "fruit",
    },
    Object {
      "name": "berry",
    },
  ],
  "updatedAt": "2018-06-20T00:00:00.000Z",
  "url": "http://localhost:8080/ipfs/QmfEcrSyPS3K7WNRZLseGu3TqSpQqRToJsvT4svUe1UDmd",
  "user": Object {
    "avatar": null,
    "displayName": "Alexandre",
    "likecoinId": "likecoin_id_1",
    "wallet": "address 1",
  },
  "wallet": "address 1",
  "width": 20,
}
`;

exports[`get featured assets 1`] = `
Array [
  Object {
    "createdAt": "2018-06-20T00:50:00.000Z",
    "description": "feijoa",
    "fingerprint": "cbc6631a005b10dcd9f45c86e0daa9a95319e568d85bfd5749d010cee1f05ecf",
    "height": 150,
    "ipfs": "QmWpZvgF4j9RuoPrj7o7xs6f8akDm9iGXBnQiUCqQw95xF",
    "ipld": "zdpuAmpgiNUXWxpJALeJi9hfjdzvNFRkUu8ptEvaoAZm6zbEb",
    "license": "cc-by",
    "like": Object {
      "count": 50,
    },
    "tags": Array [
      Object {
        "name": "feijoa",
      },
      Object {
        "name": "fruit",
      },
    ],
    "updatedAt": "2018-06-20T00:50:00.000Z",
    "url": "http://localhost:8080/ipfs/QmWpZvgF4j9RuoPrj7o7xs6f8akDm9iGXBnQiUCqQw95xF",
    "user": Object {
      "avatar": null,
      "displayName": "Sébastien",
      "likecoinId": "likecoin_id_3",
      "wallet": "address 3",
    },
    "wallet": "address 3",
    "width": 200,
  },
  Object {
    "createdAt": "2018-06-20T00:40:00.000Z",
    "description": "elderberry",
    "fingerprint": "706109d9762f6304f1c2fcc603bb2435efebb041d07459f2fc90e410917e1bdc",
    "height": 300,
    "ipfs": "QmWLPCfEQMLDY5nxM3NGEFjQBHucLmGPmRU72R6RNUDwiC",
    "ipld": "zdpuB31BxhWgsdidbhMxawbLvguKn59WzP82sU4zKU2kzcHBj",
    "license": "cc0",
    "like": Object {
      "count": 40,
    },
    "tags": Array [
      Object {
        "name": "elderberry",
      },
      Object {
        "name": "fruit",
      },
    ],
    "updatedAt": "2018-06-20T00:40:00.000Z",
    "url": "http://localhost:8080/ipfs/QmWLPCfEQMLDY5nxM3NGEFjQBHucLmGPmRU72R6RNUDwiC",
    "user": Object {
      "avatar": null,
      "displayName": "Pooya",
      "likecoinId": "likecoin_id_2",
      "wallet": "address 2",
    },
    "wallet": "address 2",
    "width": 400,
  },
  Object {
    "createdAt": "2018-06-20T00:30:00.000Z",
    "description": "dragonfruit",
    "fingerprint": "b53322cf5c9d946fcf796e36ca5a8c3f921241cc0e34565c1471d7ac142403af",
    "height": 150,
    "ipfs": "QmaADwDYHmkJrYcPyVF3YWFkuwYWB1VyhEtdefn1TpHowt",
    "ipld": "zdpuAn5TQWWNZxG1kfgSq3v3f7wEe3pTJEVQjdhDhM3ebm3Fr",
    "license": "cc-by-nd",
    "like": Object {
      "count": 30,
    },
    "tags": Array [
      Object {
        "name": "fruit",
      },
      Object {
        "name": "berry",
      },
      Object {
        "name": "dragonfruit",
      },
    ],
    "updatedAt": "2018-06-20T00:30:00.000Z",
    "url": "http://localhost:8080/ipfs/QmaADwDYHmkJrYcPyVF3YWFkuwYWB1VyhEtdefn1TpHowt",
    "user": Object {
      "avatar": null,
      "displayName": "Alexandre",
      "likecoinId": "likecoin_id_1",
      "wallet": "address 1",
    },
    "wallet": "address 1",
    "width": 200,
  },
  Object {
    "createdAt": "2018-06-20T01:30:00.000Z",
    "description": "kiwifruit",
    "fingerprint": "b345cbc239feb011fb24640b75db37e4a329f7485f5d5331c3fedbbd74b54864",
    "height": 200,
    "ipfs": "QmVj5TG7RbzgVEgEuECoqZ1cucvYwPasy6YduAs2Shmsv5",
    "ipld": "zdpuAwvtnzskcuS9t4dN8FHnZ8RdPWYSHWCHTCkdtTPTNLCx5",
    "license": "cc-by",
    "like": Object {
      "count": 30,
    },
    "tags": Array [
      Object {
        "name": "kiwifruit",
      },
      Object {
        "name": "fruit",
      },
    ],
    "updatedAt": "2018-06-20T01:30:00.000Z",
    "url": "http://localhost:8080/ipfs/QmVj5TG7RbzgVEgEuECoqZ1cucvYwPasy6YduAs2Shmsv5",
    "user": Object {
      "avatar": null,
      "displayName": "Alexandre",
      "likecoinId": "likecoin_id_1",
      "wallet": "address 1",
    },
    "wallet": "address 1",
    "width": 150,
  },
  Object {
    "createdAt": "2018-06-20T00:20:00.000Z",
    "description": "cherry",
    "fingerprint": "8b0d0ba9e502d6e39771e129076a605036b4871cefa986352993e736495803ce",
    "height": 300,
    "ipfs": "QmTtSSQyJ5HyCAo9k478SapXQ7anf4YxEjhsRz1KuZdJBN",
    "ipld": "zdpuAqNBiRBbEEsie4G44A3X5mSuNX3Bk1k9D5Y8b2Fmg1e33",
    "license": "cc-by-sa",
    "like": Object {
      "count": 20,
    },
    "tags": Array [
      Object {
        "name": "cherry",
      },
      Object {
        "name": "fruit",
      },
      Object {
        "name": "berry",
      },
    ],
    "updatedAt": "2018-06-20T00:20:00.000Z",
    "url": "http://localhost:8080/ipfs/QmTtSSQyJ5HyCAo9k478SapXQ7anf4YxEjhsRz1KuZdJBN",
    "user": Object {
      "avatar": null,
      "displayName": "Sébastien",
      "likecoinId": "likecoin_id_3",
      "wallet": "address 3",
    },
    "wallet": "address 3",
    "width": 400,
  },
  Object {
    "createdAt": "2018-06-20T01:20:00.000Z",
    "description": "jackfruit",
    "fingerprint": "d73adb98f680ee85f4fedd88bfa69e1928c864660b560097f41a912f30fa6e60",
    "height": 400,
    "ipfs": "Qmas9MoA9GAz5cgy9LuZqGD6iAPcKPARcaqT4wd5488jFn",
    "ipld": "zdpuAzN4Jt95XEEptW9LrUsUHxvbiJNcx316fVo2E8HPD7cfM",
    "license": "cc0",
    "like": Object {
      "count": 20,
    },
    "tags": Array [
      Object {
        "name": "jackfruit",
      },
      Object {
        "name": "fruit",
      },
    ],
    "updatedAt": "2018-06-20T01:20:00.000Z",
    "url": "http://localhost:8080/ipfs/Qmas9MoA9GAz5cgy9LuZqGD6iAPcKPARcaqT4wd5488jFn",
    "user": Object {
      "avatar": null,
      "displayName": "Sébastien",
      "likecoinId": "likecoin_id_3",
      "wallet": "address 3",
    },
    "wallet": "address 3",
    "width": 300,
  },
  Object {
    "createdAt": "2018-06-20T00:10:00.000Z",
    "description": "banana",
    "fingerprint": "537777bd7a5fc4ed91a74ef6fa7d8c26dc2385af90a9e48c0482a4b27d891b3f",
    "height": 150,
    "ipfs": "QmXfjc79upANgieA3CsaDg5bSJLGmdiHFJ4YRPtKM5ZB5m",
    "ipld": "zdpuAqxDjsBiXnDkNfj3LwLWp7EhPoej52hXtYLcuVxa1SwcY",
    "license": "cc-by",
    "like": Object {
      "count": 10,
    },
    "tags": Array [
      Object {
        "name": "fruit",
      },
      Object {
        "name": "banana",
      },
      Object {
        "name": "berry",
      },
    ],
    "updatedAt": "2018-06-20T00:10:00.000Z",
    "url": "http://localhost:8080/ipfs/QmXfjc79upANgieA3CsaDg5bSJLGmdiHFJ4YRPtKM5ZB5m",
    "user": Object {
      "avatar": null,
      "displayName": "Pooya",
      "likecoinId": "likecoin_id_2",
      "wallet": "address 2",
    },
    "wallet": "address 2",
    "width": 200,
  },
  Object {
    "createdAt": "2018-06-20T01:10:00.000Z",
    "description": "honeyberry",
    "fingerprint": "2ccbdcc246b1d2905e66b9b1917957e5da3a555b4ba3265f0c4818107afa059f",
    "height": 200,
    "ipfs": "QmPuwRC1aihWD4Cap2yYqZFQhArdfSRPfuUjVn4LKT9rtr",
    "ipld": "zdpuAwvvfbqutNn7eRUpJdwq8AYzmYGRy54fcqr5Ew72orG4o",
    "license": "cc-by-nd",
    "like": Object {
      "count": 10,
    },
    "tags": Array [
      Object {
        "name": "fruit",
      },
      Object {
        "name": "honeyberry",
      },
    ],
    "updatedAt": "2018-06-20T01:10:00.000Z",
    "url": "http://localhost:8080/ipfs/QmPuwRC1aihWD4Cap2yYqZFQhArdfSRPfuUjVn4LKT9rtr",
    "user": Object {
      "avatar": null,
      "displayName": "Pooya",
      "likecoinId": "likecoin_id_2",
      "wallet": "address 2",
    },
    "wallet": "address 2",
    "width": 150,
  },
  Object {
    "createdAt": "2018-06-20T00:00:00.000Z",
    "description": "avocado",
    "fingerprint": "04f01eb90f7be1b1628218c5a9f1f87c5704bc79548f15673695a3e7264ba88b",
    "height": 15,
    "ipfs": "QmfEcrSyPS3K7WNRZLseGu3TqSpQqRToJsvT4svUe1UDmd",
    "ipld": "zdpuAqoWEAALng3FeUcF8GY66UsK17wd3fGxr9XbXu6baoKLN",
    "license": "cc0",
    "like": Object {
      "count": 0,
    },
    "tags": Array [
      Object {
        "name": "fruit",
      },
      Object {
        "name": "berry",
      },
      Object {
        "name": "avocado",
      },
    ],
    "updatedAt": "2018-06-20T00:00:00.000Z",
    "url": "http://localhost:8080/ipfs/QmfEcrSyPS3K7WNRZLseGu3TqSpQqRToJsvT4svUe1UDmd",
    "user": Object {
      "avatar": null,
      "displayName": "Alexandre",
      "likecoinId": "likecoin_id_1",
      "wallet": "address 1",
    },
    "wallet": "address 1",
    "width": 20,
  },
  Object {
    "createdAt": "2018-06-20T01:00:00.000Z",
    "description": "grape",
    "fingerprint": "758cbe3689c4833c2219f795839b83c05f420c91088132ffc61123b5226f1a04",
    "height": 20,
    "ipfs": "Qme19Y2ZHXHatSZMH2bzXiXMzV2uau7YJwsQLrTSWgjgM8",
    "ipld": "zdpuAmh7AAL14HJTHJGbiaEB5g4EfRxZcjYfkXN55LvqPZpcN",
    "license": "cc-by-sa",
    "like": Object {
      "count": 0,
    },
    "tags": Array [
      Object {
        "name": "fruit",
      },
      Object {
        "name": "grape",
      },
    ],
    "updatedAt": "2018-06-20T01:00:00.000Z",
    "url": "http://localhost:8080/ipfs/Qme19Y2ZHXHatSZMH2bzXiXMzV2uau7YJwsQLrTSWgjgM8",
    "user": Object {
      "avatar": null,
      "displayName": "Alexandre",
      "likecoinId": "likecoin_id_1",
      "wallet": "address 1",
    },
    "wallet": "address 1",
    "width": 15,
  },
]
`;
